realm/realm-library/src/main/java/io/realm/internal/ColumnInfo.java

```diff
@@ -33,9 +33,9 @@
  * <p>
  * While the use of the fields in {@code ColumnDetails} is consistent, there are three subtly different cases:
  * <ul>
- * <li>If the column type is a simple type, the link table field is empty (0L / NULLPTR)</li>
- * <li>If the column type is OBJECT or LINK, the link table field is the class name of the OBJECT/LINK type</li>
- * <li>If the column type is LINKING_OBJECT, the link table field is the class name of the backlink source table
+ * <li>If the column type is a simple type, the {@code peerClassName} field is {@code null}</li>
```

Comment:
This is not true.  It needs to be 0L / NULLPTR.  This value is use only by native code and it must appear to the native code, to be NULLPTR.